handlers/new_session.go

```diff
@@ -9,14 +9,19 @@ import (
 )
 
 func NewSession(rw http.ResponseWriter, req *http.Request) {
-	if !services.IsHuman(req) {
+	req.ParseForm()
+	challenge := req.Form.Get("g-recaptcha-response")
+	if !services.IsHuman(challenge) {
```

Comment:
You're right,  I took the logic outside the function mainly because parse form needs to be done outside. I can still send the request anyways and parse the form in the NewSession function.

Does that sound right?